keyless/circuit-tests/src/packing.rs

```diff
@@ -0,0 +1,159 @@
+
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_keyless_common::input_processing::circuit_input_signals::CircuitInputSignals;
+use aptos_keyless_common::input_processing::config::CircuitPaddingConfig;
+use crate::TestCircuitHandle;
+use itertools::*;
+use rand_chacha::{rand_core::{SeedableRng as _, RngCore as _}, ChaCha20Rng};
+use aptos_crypto::poseidon_bn254;
+use ark_bn254::Fr;
+use ark_ff::PrimeField;
+use ark_ff::BigInteger;
+
+
+const TEST_RNG_SEED : u64 = 2513478;
+
+fn expected_num2bits_be(n: u64, size: usize) -> Vec<u8> {
+
+    let mut bits_le = Vec::new();
+    for i in 0..size {
+        bits_le.push( ( (n >> i) & 1 ) as u8 );
+    }
+    bits_le.into_iter().rev().collect()
+}
+
+fn expected_bits_to_field_elems(bits: &[bool], bits_per_field_elem : usize) -> Vec<Fr> {
+    bits.chunks(bits_per_field_elem)
+        .map(bits_to_field_elem)
+        .collect()
+}
+
+fn bits_to_field_elem(bits: &[bool]) -> Fr {
+    let bigint = BigInteger::from_bits_be(bits);
+    Fr::from_bigint(bigint).unwrap()
+}
+
+#[test]
+fn num2bits_be_test() {
+    let circuit_handle = TestCircuitHandle::new("packing/num2bits_be_test.circom").unwrap();
+    let bits_max_size = 8;
+
+
+    for n in 0..=255 {
+        let config = CircuitPaddingConfig::new()
+            .max_length("bits_out", bits_max_size);
+
+        let circuit_input_signals = CircuitInputSignals::new()
+            .byte_input("num_in", n)
+            .bytes_input("bits_out", &expected_num2bits_be(n as u64, bits_max_size))
```

Comment:
bit: use the `bits_input` introduced and update the return type of `expected_num2bits_be`.